src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -352,27 +379,11 @@ const createSearchSourceWithQuery = async (
     .setField('filter', timeRangeFilter ? [timeRangeFilter] : []);
 
   // Add histogram aggregations if requested and time-based
-  if (includeHistogram && indexPattern.timeFieldName) {
-    const timeRange = services.data.query.timefilter.timefilter.getTime();
-    const aggConfig = {
-      aggs: {
-        histogram: {
-          date_histogram: {
-            field: indexPattern.timeFieldName,
-            interval: customInterval || 'auto',
-            min_doc_count: 0,
-            extended_bounds: {
-              min: timeRange.from,
-              max: timeRange.to,
-            },
-          },
-        },
-      },
-      size: 500, // Limit hits for performance
-    };
-
-    searchSource.setField('aggs', aggConfig.aggs);
-    searchSource.setField('size', aggConfig.size);
+  if (includeHistogram && indexPattern.timeFieldName && customInterval) {
+    const dslAggs = createHistogramConfigs(indexPattern, customInterval, services.data)!.toDsl();
+    searchSource.setField('aggs', dslAggs);
+    // TODO: Do we want to hard code it to 500?
+    searchSource.setField('size', 500);
```

Comment:
Let's avoid non-null assertion
```suggestion
    const histogramConfigs = createHistogramConfigs(indexPattern, customInterval, services.data);
    if (histogramConfigs) {
      const dslAggs = histogramConfigs.toDsl();
      searchSource.setField('aggs', dslAggs);
      // TODO: Do we want to hard code it to 500?
      searchSource.setField('size', 500);
    }
```